python_modules/dagster/dagster/_core/definitions/partition_mapping.py

```diff
@@ -215,6 +219,107 @@ def get_downstream_partitions_for_partition_range(
         raise NotImplementedError()
 
 
+@experimental
+class SingleDimensionDependencyMapping(PartitionMapping):
+    """
+    Defines a correspondence between an upstream single-dimensional partitions definition
+    and a downstream MultiPartitionsDefinition. The upstream partitions definition must be
+    a dimension of the downstream MultiPartitionsDefinition.
+
+    For an upstream partition key X, this partition mapping assumes that any downstream
+    multi-partition key with X in the selected dimension is a downstream dependency.
+
+    Args:
+        partition_dimension_name (str): The name of the partition dimension in the downstream
+            MultiPartitionsDefinition that matches the upstream partitions definition.
+    """
+
+    def __init__(self, partition_dimension_name: str):
+        self.partition_dimension_name = partition_dimension_name
+
+    def _check_upstream_partitions_def_equals_selected_dimension(
+        self,
+        upstream_partitions_def: PartitionsDefinition,
+        downstream_partitions_def: MultiPartitionsDefinition,
+    ) -> None:
+        matching_dimensions = [
+            partitions_def
+            for partitions_def in downstream_partitions_def.partitions_defs
+            if partitions_def.name == self.partition_dimension_name
+        ]
+        if len(matching_dimensions) != 1:
+            check.failed(f"Partition dimension '{self.partition_dimension_name}' not found")
+        matching_dimension_def = next(iter(matching_dimensions))
+
+        if upstream_partitions_def != matching_dimension_def.partitions_def:
```

Comment:
Something to keep in mind for future optimization is that this `!=` could be expensive for large `StaticPartitionsDefinition`s.